name: Validate

on: pull_request

jobs:
  unit-test:
    name: Unit test ðŸ§ª
    runs-on: ubuntu-latest
    steps:
      # Fix for NX error "fatal: No such ref: 'origin/main'"
      # https://github.com/nrwl/nx/issues/2170#issuecomment-609786474
      - name: Checkout ðŸ›Ž
        uses: actions/checkout@v2
        with:
          ref: ${{ github.event.pull_request.head.ref }}
          fetch-depth: 0

      - name: Setup node ðŸ› 
        uses: actions/setup-node@v2
        with:
          node-version: '16'
          cache: yarn

      - run: git fetch --no-tags --prune --depth=5 origin/master

      - name: Install dependencies ðŸ“¦
        run: yarn install --frozen-lockfile

      - name: Run unit tests on affected packages ðŸ”„
        run: yarn nx affected:test --base=origin/$GITHUB_BASE_REF
        env:
          CI: true
